src/components/Frame/components/Loading/Loading.tsx

```diff
@@ -1,79 +1,41 @@
-import React, {Component} from 'react';
-import debounce from 'lodash/debounce';
+import React, {useEffect, useState} from 'react';
 
-import styles from './Loading.scss';
-
-export interface LoadingProps {}
+import {useI18n} from '../../../../utilities/i18n';
 
-interface State {
-  progress: number;
-  step: number;
-  animation: number | null;
-}
+import styles from './Loading.scss';
 
 const INITIAL_STEP = 10;
 const STUCK_THRESHOLD = 99;
 
-export class Loading extends Component<LoadingProps, State> {
-  state: State = {
-    progress: 0,
-    step: INITIAL_STEP,
-    animation: null,
-  };
-
-  private ariaValuenow = debounce(() => {
-    const {progress} = this.state;
-    return Math.floor(progress / 10) * 10;
-  }, 15);
-
-  componentDidMount() {
-    this.increment();
-  }
-
-  componentWillUnmount() {
-    const {animation} = this.state;
+export function Loading() {
+  const i18n = useI18n();
 
-    if (animation != null) {
-      cancelAnimationFrame(animation);
-    }
-  }
-
-  render() {
-    const {progress} = this.state;
-
-    const customStyles = {
-      transform: `scaleX(${Math.floor(progress) / 100})`,
-    };
-
-    const ariaValuenow = this.ariaValuenow();
-
-    return (
-      <div className={styles.Loading}>
-        <div
-          className={styles.Level}
-          style={customStyles}
-          aria-valuenow={ariaValuenow}
-          aria-valuemin={0}
-          aria-valuemax={100}
-          role="progressbar"
-        />
-      </div>
-    );
-  }
-
-  private increment() {
-    const {progress, step} = this.state;
+  const [progress, setProgress] = useState(0);
+  const [step, setStep] = useState(INITIAL_STEP);
 
+  useEffect(() => {
     if (progress >= STUCK_THRESHOLD) {
       return;
     }
 
-    const animation = requestAnimationFrame(() => this.increment());
```

Comment:
I don't know why we are using `requestAnimationFrame` for incrementing the [value on a transform](https://github.com/Shopify/polaris-react/pull/3770/files#diff-21fb7fa504ae13f32acd177042514b61f7cde459c035fe20b4879d54c1fd1cceL45). From my understanding the transform should be doing transitions on the GPU so we wouldn't be getting a performance improvement by waiting for the next frame.